pkgs/development/tools/analysis/retdec/default.nix

```diff
@@ -169,68 +187,105 @@ stdenv.mkDerivation (self: {
 
-      substituteInPlace $gtest \
-        --replace-fail '$'{GTEST_LIB} "GTest::gtest"\
-        --replace-fail '$'{GMOCK_LIB} "GTest::gmock"\
-        --replace-fail '$'{GTEST_MAIN_LIB} "GTest::gtest_main"\
-        --replace-fail '$'{GMOCK_MAIN_LIB} "GTest::gmock_main"
+      substituteInPlace src/utils/CMakeLists.txt \
+        --replace-warn '$'{RETDEC_GIT_VERSION_TAG} ${self.version} \
+        --replace-warn '$'{RETDEC_GIT_COMMIT_HASH} ${self.src.rev}
 
-      # without git history, there is no chance these tests will pass.
+
+      # tests: without git history, there is no chance these tests will pass.
       substituteInPlace tests/utils/version_tests.cpp \
-        --replace-quiet VersionTests DISABLED_VersionTests
+        --replace-warn VersionTests DISABLED_VersionTests
+
 
+      # scripts: patch paths
       substituteInPlace scripts/retdec-utils.py \
-        --replace-warn /usr/bin/time ${time} \
-        --replace-warn /usr/local/bin/gtime ${time}
+        --replace-fail /usr/bin/time ${time} \
+        --replace-fail /usr/local/bin/gtime ${time}
       substituteInPlace scripts/retdec-unpacker.py \
-        --replace-warn "'upx'" "'${upx}'"
+        --replace-fail "'upx'" "'${upx}'"
     '';
 
   doInstallCheck = enableTests;
   installCheckPhase = ''
-    ${python3.interpreter} "$out/bin/retdec-tests-runner.py"
+    ${python3.interpreter} "''${!outputBin}/bin/retdec-tests-runner.py"
 
     rm -rf $out/bin/__pycache__
   '';
 
+
+  # static-code patterns are split out since they are rarely needed by end users.
+  # from retdec's wiki:
+  #   However, these are mainly for older compilers used for testing purposes.
+  #   If you want to remove code from newer compilers or some custom libraries,
+  #   you have to create your own signature files.
+  # see: https://github.com/avast/retdec/wiki/Removing-Statically-Linked-Code
+  postFixup = ''
+    mkdir -p $dev/lib/retdec
+    mv -v $out/share/retdec/cmake $dev/lib/retdec
+
+    yara_dir=share/retdec/support/generic/yara_patterns
+    mkdir -p $patterns/$yara_dir
+    mv -v $out/$yara_dir/static-code $patterns/$yara_dir
+  '';
+
+  outputs = [ "out" "lib" "dev" "patterns" ];
+
+  passthru.full = buildEnv {
+    name = "${self.pname}-full-${self.version}";
+    inherit (self.finalPackage) meta;
+    paths = [ self.finalPackage.out self.finalPackage.patterns ];
+    postBuild = ''
+      config=$out/share/retdec/decompiler-config.json
+      cp -v --remove-destination $(readlink -f $config) $config
+
+      static_code=support/generic/yara_patterns/static-code/
+      substituteInPlace $config \
+        --replace-fail ./$static_code $out/share/retdec/$static_code
+    '';
+  };
+
+  passthru.deps = deps;
+
   meta = with lib; {
     description = "Retargetable machine-code decompiler based on LLVM";
     homepage = "https://retdec.com";
     license = licenses.mit;
     maintainers = with maintainers; [ katrinafyi ];
-    platforms = [ "x86_64-linux" ];
   };
```

Comment:
`platforms` should be set, even if it’s to `lib.platforms.all` (though I’d be shocked if this derivation is that portable).